pkg/core/resources/apis/meshservice/generate/generator_test.go

```diff
@@ -266,20 +266,25 @@ var _ = Describe("MeshService generator", func() {
 		dp := core_mesh.NewDataplaneResource()
 		Expect(resManager.Delete(context.Background(), dp, store.DeleteByKey("dp-1", model.DefaultMesh))).To(Succeed())
 
+		gracePeriodStartedAt := ""
 		// Wait until the MeshService has been marked with grace period start
 		Eventually(func(g Gomega) {
 			g.Expect(resManager.Get(context.Background(), ms, store.GetByKey("backend", model.DefaultMesh))).To(Succeed())
-			g.Expect(ms.GetMeta().GetLabels()).To(HaveKey("kuma.io/deletion-grace-period-started-at"))
+			g.Expect(ms.GetMeta().GetLabels()).To(HaveKey(mesh_proto.DeletionGracePeriodStartedLabel))
+			gracePeriodStartedAt = ms.GetMeta().GetLabels()[mesh_proto.DeletionGracePeriodStartedLabel]
 		}, "2s", "100ms").Should(Succeed())
+
 		labelExistedSince := time.Now()
+		err = labelExistedSince.UnmarshalText([]byte(gracePeriodStartedAt))
+		Expect(err).ToNot(HaveOccurred())
 
 		// Before the grace period it still exists and afterwards it eventually
 		// disappears
 		Consistently(func(g Gomega) {
 			if time.Now().Before(labelExistedSince.Add(gracePeriodInterval)) {
```

Comment:
Ok so `labelExistedSince` is going to be _earlier_, which makes it less likely that the `Before` check will succeed and the branch will be taken. Seems to be an improvement.

Should we maybe only call `Expect` _after_ we call `Get` and only if the time is still `Before`? Cause AFAICT we still have the possibility of checking `Before` but then the resource is deleted, and the Consistently fails